doc/website/getting-started/overview.md

```diff
@@ -0,0 +1,540 @@
+
+We distinguish between the ``typed API`` and the ``untyped API``. In the Typed API the underlying data type is made
+apparent by typed pointers or references to some data type T (often a template parameter). this allows working with
+the data in an C++ idiomatic and type-safe way and should be preferred whenever possible.
+
+The Untyped API provides opaque (i.e. void) pointers to data, which is flexible and efficient but also requires that
+the user takes care to interpret received data correctly, i.e. as a type compatible to what was actually sent. This
+is required for interaction with other lower level APIs and should be used sparingly. For further information see the
+respective header files.
+
+There also is a plain [C API](../../../iceoryx_examples/icedelivery_on_c/README.md), which can be used if C++ is not
+an option.
+
+We now describe the how to use the API in iceoryx applications. We will omit namespaces in several places to keep
+the code concise. In most cases it can be assumed that we are using namespace ``iox::cxx``. We also will use ``auto``
+sparingly to clearly show which types are involved, but in many cases automatic type deduction is possible and can
+shorten the code.
+
+### Optional
+
+The type ``cxx::optional<T>`` is used to indicate that there may or may not be a value of a specific type ``T``
+available. This is essentially the maybe monad in functional programming. Assuming we have some optional (usually the
+result of some computation)
+```
+optional<int> result = someComputation();
+```
+we can check for its value using
+```
+if(result.has_value())
+{
+    auto value = result.value();
+    //do something with the value
+} else {
+    //handle the case that there is no value
+}
+```
+A shorthand to get the value is 
+```
+auto value = *result;
+```
+
+Note that getting the value if there is no value is undefined behavior, so it must be checked beforehand.
+
+We can achieve the same with the functional approach by providing a function for both cases.
+
+```
+result.and_then([](int& value) { /*do something with the value*/ })
+      .or_else([]() { /*handle the case that there is no value*/ });
+```
+Notice that we get the value by reference, so if a copy is desired it has to be created explicitly in the lambda or
+function we pass.
+
+The optional can be be initialized from a value directly
+```
+optional<int> result = 73;
+result = 37;
+```
+If it is default initialized it is automatically set to its null value of type ``iox::cxx::nullopt_t``;
+This can be also done directly by using the constant ``iox::cxx::nullopt``
+
+```
+result = iox::cxx::nullopt;
+```
+
+### Expected
+``iox::cxx::expected<T, E>`` generalizes ``iox::cxx::optional`` by admitting a value of another type ``E`` instead of
+no value at all, i.e. it contains either a value of type ``T`` or ``E`` (and roughly corresponds to the either
+monad). This is usually used to pass a value of type ``T`` or an error that may have occurred, i.e. ``E`` is the
```

Comment:
> (and roughly corresponds to the either monad)
Please either remove the braces or remove the line.

Reply:
Put the info into a new sentence.